src/ui_utils.c

```diff
@@ -2096,6 +2096,33 @@ void ui_table_add_row(GtkTable *table, gint row, ...)
 }
 
 
+/* comment-out all lines that are not already commented out except sections */
+static void comment_conf_files(ScintillaObject *sci)
+{
+	gint line, line_count;
+
+	line_count = sci_get_line_count(sci);
+	for (line = 0; line < line_count - 1; line++)
+	{
+		gint pos_start = sci_get_position_from_line(sci, line);
+		gint pos_end = sci_get_position_from_line(sci, line + 1);
+		gint pos;
+
+		for (pos = pos_start; pos < pos_end; pos++)
+		{
+			gchar c = sci_get_char_at(sci, pos);
+			if (c == '[')
+				break;
+			if (!isspace(c))
+			{
+				sci_insert_text(sci, pos_start, "#");
+				break;
+			}
+		}
```

Comment:
Yeah, something like that, but again, your code doesn't detect empty lines where we shouldn't insert `#` which is the purpose of the `isspace()` test in my code. I also think we don't really have to check for the indented `[` or `#` since these aren't valid conf files.

Reply:
>  but again, your code doesn't detect empty lines where we shouldn't insert `#` which is the purpose of the `isspace()` test in my code.

The intent of `if (pos_indent < pos_end)` is to detect that (again, not tested whether that actually works, and what those positions actually are)

>  I also think we don't really have to check for the indented `[` or `#` since these aren't valid conf files.

Aren't they?  So you say the *only* case that matters for checking whitespaces is whether it makes a blank line?
That would make the code look like that:

```c
		if (pos_indent < pos_end)
		{
			gchar c = sci_get_char_at(sci, pos_start);
			if (c != '#' && c != '[')
				sci_insert_text(sci, pos_start, "#");
		}
```

Or indeed your loop, whichever you like best.

Reply:
> I also think we don't really have to check for the indented `[` or `#` since these aren't valid conf files.

I just checked and they actually [are valid per `GKeyFile` handling](https://gitlab.gnome.org/GNOME/glib/-/blob/main/glib/gkeyfile.c#L1277).

Reply:
It's actually a simpler code when we also detect the indented `[` and `#` so let's leave it that way. Interesting that it's valid for GKeyFile - they are definitely not valid for the Scintilla lexer.

I just re-pushed the fix for the patch - I used the `sci_get_line_end_position()` which is definitely better and removed the comment for `#` as the decision seems to be not to have the `##` comments for the original comments. The rest is just my original patch - I find it simple enough but maybe there's a better way to write it or there's maybe something I'm still overlooking.